### ISSUE TITLE
Analytics - Confusing event action triggers

### ISSUE THREAD

**Issue by Bernietorras** [GoogleForCreators/web-stories-wp#4865]
## Bug Description

1- When inside a story you go to the share button, it triggers the event action: story_open and story_close at the same time (seems wrong).

2- When you are at last story, the story-last-page-visible event action is not triggered.

## Expected Behaviour

1- When you click the share button, only the story_open event action should be triggered. The story_close action should be triggered only when you click on it to close the dialog.

2- 

a. When you are close to finish and opens the last story you expect this event action: story-last-page-visible.

b. When the story is finished, the event action open_story is triggered, is not wrong but confusing, in this case it could trigger the story-bookend-enter instead of story_open. This way we could leave open_story only for share button, so we can separate both actions and have a clearer metric.

## Steps to Reproduce

I have measured through real time analytics the events that have been triggered. The navigation through the story has been done through a mobile device.

## Screenshots

1- Events triggering at the same time: https://www.bernietorras.com/wp-content/uploads/2020/10/share-link-event-issue.jpg

2- Finished story: https://www.bernietorras.com/wp-content/uploads/2020/10/finished_story.jpg

## Additional Context

https://github.com/ampproject/amphtml/blob/master/extensions/amp-story/amp-story-analytics.md

Story used for the example: https://www.bernietorras.com/web-stories/web-stories-meetup/

- Plugin Version: 1.0.0 installed from the WordPress repository
- Operating System: Win 10 & Android
- Browser: Chrome 86

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by swissspidy** [comment]
Heya, thanks for your report.

I just did some testing with the help of [this extension](https://chrome.google.com/webstore/detail/google-analytics-and-segm/ppknjgihbmfcdmgdenblfomidfomhclp).

> When inside a story you go to the share button, it triggers the event action: story_open and story_close at the same time

`story_open` is fired when opening a drawer or dialog inside a story (e.g. page attachment). And `story_close` is fired when closing. The share dialog is such a drawer.

In my testing the two events didn't trigger simultaneously. So I cannot reproduce what you're seeing.

I navigate through the story, and on the last page I tap to advance to open the sharing dialog -> `story_open` fires. I copy the URL -> nothing fires. I tap on the story to close the dialog -> `story_close` fires.

> When you are at last story, the `story-last-page-visible` event action is not triggered.

It is triggered just fine in my testing and I can also see it in your screenshots. The event name is "story_complete" (`story-last-page-visible` is just the AMP-internal name, not the event name you see in GA)

---- 

I think what would help here is triggering story-bookend-enter / story-bookend-exit / story-bookend-click events for the bookend shown at the end.

We didn't include these originally because it's eventually being deprecated. But since it's still there at the moment, I think it makes sense to add that.

**Comment by Bernietorras** [comment]
Hey Pascal!! Thank you very much for your quick response!

You are right. From desktop there's no problem, the story_close event is fired correctly in share button. 

I find the problem about share button when I launch it from my mobile device. (Huawei P20).

I thought it would be a problem with my device, when I remembered that I have Android Studio installed and I have released a virtual device of Google Pixel XL. So I used it to test, I trust more than one Google device, even if it is virtual. ;)

You will see in the video I recorded that it still gives this problem, both events are activated when I click on the share button:
https://www.loom.com/share/c7d07d37c8d34faebc4b1c478bc9f88d

Thank you for considering my insights!
